packages/html-reporter/src/index.tsx

```diff
@@ -14,14 +14,50 @@
  * limitations under the License.
  */
 
-import type { HTMLReport } from '@playwright/test/src/reporters/html';
+import type { HTMLReport, TestAttachment } from '@playwright/test/src/reporters/html';
 import type zip from '@zip.js/zip.js';
 import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import './colors.css';
 import { LoadedReport } from './loadedReport';
 import { ReportView } from './reportView';
 
+export type Metadata = Partial<{
+  'generatedAt': number;
+  'revision.id': string;
+  'revision.author': string;
+  'revision.email': string;
+  'revision.subject': string;
+  'revision.timestamp': number;
+  'revision.link': string;
+  'revision.localPendingChanges': boolean;
+  'ci.link': string;
+}>;
+
+const extractMetadata = (attachments: TestAttachment[]): Metadata | undefined => {
```

Comment:
This is a result of deciding to do an attachment per-field (instead of the previous iteration where we just put everything in one attachment and just did JSON.parse(…) once here).

We chose one-field-per-file so "plugins" don't have to be concerned with merging.